ipatests/test_integration/test_installation.py

```diff
@@ -479,3 +481,51 @@ def test_reserved_ip_as_forwarder(self):
         exp_str = ("Invalid IP Address 0.0.0.0: cannot use IANA reserved "
                    "IP address 0.0.0.0")
         assert exp_str in cmd.stdout_text
+
+
+class TestKRAinstallAfterCertRenew(IntegrationTest):
+    """ Test KRA installtion after ca agent cert renewal
+
+    KRA installtion was failing after ca-agent cert gets renewed.
+    This test check if the syptoms no longer exists.
+
+    related ticket: https://pagure.io/freeipa/issue/7288
+    """
+
+    def test_KRA_install_after_cert_renew(self):
+
+        tasks.install_master(self.master)
+
+        # get ca-agent cert contents
+        p12_content = self.master.get_file_contents(paths.DOGTAG_ADMIN_P12)
+        admin_pass = self.master.config.admin_password
+
+        # convert cert to pem format
+        p12 = crypto.load_pkcs12(p12_content, admin_pass)
+        cert = crypto.dump_certificate(crypto.FILETYPE_PEM,
+                                       p12.get_certificate())
+
+        # get expiry date of agent cert
+        x509 = crypto.load_certificate(crypto.FILETYPE_PEM, cert)
+        cert_expiry = x509.get_notAfter()
+
+        # remove the 'Z' from the date string
+        cert_expiry = cert_expiry[:-1].decode('utf-8')
+        py_date_obj = datetime.strptime(cert_expiry, "%Y%m%d%H%M%S")
+
```

Comment:
Is this needed? I know, Z could look strange, but it is zero, it means UTC. I searched for it and I found out:
https://stackoverflow.com/questions/19654578/python-utc-datetime-objects-iso-format-doesnt-include-z-zulu-or-zero-offset

Reply:
The expiry date was having Z in the string and was failing my side. That's why this workaround.

Reply:
@mrizwan93  What about this :-)
```
py_date_obj = datetime.strptime(date, "%Y%m%d%H%M%SZ")
```
Just add Z to the format string, then you can remove line '512' and '513'.

There is an example that it is working:
```
date = "20131029093841Z"
py_date_obj = datetime.strptime(date, "%Y%m%d%H%M%SZ")
print(py_date_obj)
```

Reply:
It does not matter, use python-cryptography and you don't have to do any of this at all.